### ISSUE TITLE
Add note about JADE -> PUG

### ISSUE THREAD

**Issue by teodors** [pugjs/pug#2309]
I like the new name, but I was a bit confused. Lots of articles and people link to this repository as Jade. When you get here it is Pug. You should really put a big bolded warning on top of README.md that Jade is now Pug. This would solve a lot of confusion.

**Comment by andyspurgeon** [comment]
Agree.  This is particularly confusing when the pug website (pug-lang) appears to just be a placeholder and other links in the README (e.g. the test drive) do not work.  It makes the project appear dead.

**Comment by fritx** [comment]
What's going on here?

**Comment by icflorescu** [comment]
@fritx - looks like some patent trolls from jadeworld dot com are claiming ownership over the word "jade". Most likely because they have no chance of surpassing jade-lang on Google search. What they failed to realise is that the whole scheme will eventually turn against them, since they certainly managed to alienate lots of people. I sure hope somebody from another industry will do the same to them at some point, because in the end, you know, jade is a mineral compound. 

This kind of reminds me of Microsoft trying to claim ownership over "windows" years ago.
I wonder what's going to happen when some smart-ass lawyer is going to do the same for "script".

Meanwhile, I'll probably have to change the name of the old [aspax-jade-handler](https://github.com/icflorescu/aspax-jade-handler) plugin. There are so many jade-based packages in the Node.js ecosystem ([jadeify](https://github.com/domenic/jadeify), the webpack [jade loader](https://github.com/webpack/jade-loader), [fly-jade](https://github.com/sugarshin/fly-jade), etc.), I don't even want to think about the implications...

@ForbesLindesay - please let me know if I can help with the transition.

**Comment by dasilvacontin** [comment]
At this point one starts considering to register the library's name after 20 commits or so...

**Comment by udp** [comment]
Repo description needs updating too.

**Comment by cfresh** [comment]
Also, none of the links on the README work; pug-lang.com is unregistered, the demo link gives a 404.

**Comment by teodors** [comment]
@icflorescu @dasilvacontin This is going to sting for a very long time. For that reason, I would rather support the idea to lawyer the fudge up. I would pitch in and think a lot of people would too. If there even exists possibility of solving it that way.

That being said, this is not the place to discuss that. This thread shouldn't grow with unrelated things.

**Comment by brandoncarl** [comment]
@ForbesLindesay @teodors Has anyone spoken with jadeworld about a compromise wherein "jade-lang" has a big link to jadeworld ("thank you for sponsoring our name).

I say this because jade-lang gets something to the effect of 20x the traffic that jade world does. This would be very cheap advertising for them and would maintain better brand appeal amongst the developer community, which, given their line of work, should be important.

I know that the "pug" transition is in progress, but it has the ability to really mess up a lot of code around the world. Sorry for the late comment, I'm late to this game.

**Comment by ianva** [comment]
It's bed name，extension name like png

**Comment by gzhjs** [comment]
lots of packages depend on pug with the name of "jade", chaning the name has caused lots of problems

**Comment by visualcookie** [comment]
@icflorescu + there's also a town in Germany called Jade and a smartphone called Jade Primo. I guess, it's like you said: they couldn't keep up on Google with their shitty stuff they do and now claim that they are the owners of the name JADE. I guess, we should all trademark all minerals on this planet and someone should also claim the word water. Just in case. 😆

**Comment by brandoncarl** [comment]
@visualcookie Right, but the point is that since "Jade" is a generic word (a gemstone), the only trademarks that matter are those that are in competing sphere (in this case a software language).

That narrows down to the single company – just wondering if they'd be open to some free marketing for themselves in return for sharing the name!

**Comment by icflorescu** [comment]
@brandoncarl - trademarking a generic word in a specific domain may be legal in some jurisdictions, but let's agree to admit it's kind of immoral and very much against common-sense. And when we have (and accept having) laws and lawyers contrasting with common-sense... well, it's the kind of thing that makes you wonder "where did it all go wrong?..."

Going back to that single company – I think the wrong has already been done and there's little they can do now to undo it. It's hard to come around and keep a good face after they already sent legally threatening letters. I, for one, would never trust working/collaborating with them and I'm certainly not the only one feeling like this.

So, let's stop debating whether pug is a "good name" or not. The resonance is always subjective and not that important. I always thought LinkedIn's "dust" was an awful choice, but didn't stop me from using it when it made sense.

Bottom line is Jade HAS been renamed, so let's spread the word and make sure people find out about this so they can switch from `jade` npm to `pug` or `pug-cli` in order to use the new features.

**Comment by brandoncarl** [comment]
@icflorescu Yes, debating whether "pug" is a good name is senseless. What's not senseless though, is recognizing that millions of lines of code will potentially be breaking as a result of this.

As a result, even changing the "official" name to "jadejs" would allow the npm package to continue to exist, and therefore prevent breaking changes across the hundreds of packages that were dependent.

The bottom line is that the name has not been changed almost anywhere except for the github repo. The website on the repo still points to jade, etc.

I'm not trying to ruffle feathers, but a change such as "jadejs" would allow the ecosystem to remain untouched. Just making sure that all viable options are considered.

**Comment by ForbesLindesay** [comment]
@brandoncarl The npm package will continue to exist indefinitely, but was deprecated as per the agreement I signed.  There will be no new versions published under the name jade (including security fixes) but there should be fairly minimal disruption caused by the name change.  The only things remaining are the website, brjade and react-jade/then-jade.

I'll add a note to the README to state that the project was formerly known as Jade.

**Comment by brandoncarl** [comment]
Got it @ForbesLindesay - sorry you had to deal with that.
